unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/VoiceChatHUD/VoiceChatWindowController.cs

```diff
@@ -0,0 +1,436 @@
+        }
+
+        SetWhichPlayerIsTalking();
+    }
+
+    public void SetVoiceChatRecording(bool recording) 
+    { 
+        voiceChatBarView.PlayVoiceChatRecordingAnimation(recording);
+        isOwnPLayerTalking = recording;
+        SetWhichPlayerIsTalking();
+    }
+
+    public void SetVoiceChatEnabledByScene(bool enabled) { voiceChatBarView.SetVoiceChatEnabledByScene(enabled); }
+
+    public void Dispose()
+    {
+        ReportMuteStatuses();
+
+        if (updateMuteStatusRoutine != null)
+            CoroutineStarter.Stop(updateMuteStatusRoutine);
+
+        voiceChatWindowView.OnClose -= CloseView;
+        voiceChatWindowView.OnJoinVoiceChat -= JoinVoiceChat;
+        voiceChatWindowView.OnGoToCrowd -= GoToCrowd;
+        voiceChatWindowView.OnMuteAll -= OnMuteAllToggled;
+        voiceChatWindowView.OnAllowUsersFilterChange -= ChangeAllowUsersFilter;
+        voiceChatBarView.OnLeaveVoiceChat -= LeaveVoiceChat;
+        dataStore.player.otherPlayers.OnAdded -= OnOtherPlayersStatusAdded;
+        dataStore.player.otherPlayers.OnRemoved -= OnOtherPlayerStatusRemoved;
+        ownProfile.OnUpdate -= OnUserProfileUpdated;
+        friendsController.OnUpdateFriendship -= OnUpdateFriendship;
+        settings.generalSettings.OnChanged -= OnSettingsChanged;
+
+        currentPlayers.Clear();
+        playersPool.Clear();
+    }
+
+    internal void CloseView() { SetVisibility(false); }
+
+    internal void JoinVoiceChat(bool isJoined)
+    { 
+        using (var iterator = currentPlayers.GetEnumerator())
+        {
+            while (iterator.MoveNext())
+            {
+                iterator.Current.Value.SetAsJoined(isJoined);
+            }
+        }
+
+        voiceChatWindowView.SetAsJoined(isJoined);
+
+        if (isJoined)
+        {
+            voiceChatBarView.Show();
+            SetWhichPlayerIsTalking();
+        }
+        else
+        {
+            dataStore.voiceChat.isRecording.Set(new KeyValuePair<bool, bool>(false, false), true);
+            isOwnPLayerTalking = false;
+            voiceChatBarView.Hide();
+        }
+
+        this.isJoined = isJoined;
+
+        if (!isJoined)
+        {
+            MuteAll(true);
+            socialAnalytics.SendVoiceChannelDisconnection();
+        }
+        else
+        {
+            MuteAll(voiceChatWindowView.isMuteAllOn);
+            socialAnalytics.SendVoiceChannelConnection(currentPlayers.Count);
+        }
+
+        dataStore.voiceChat.isJoinedToVoiceChat.Set(isJoined);
+    }
+
+    internal void LeaveVoiceChat() { JoinVoiceChat(false); }
+
+    internal void GoToCrowd() { WebInterface.GoToCrowd(); }
+
+    internal void OnOtherPlayersStatusAdded(string userId, Player player)
+    {
+        if (!currentPlayers.ContainsKey(player.id))
+        {
+            var otherProfile = UserProfileController.userProfilesCatalog.Get(player.id);
+
+            if (otherProfile != null)
+            {
+                VoiceChatPlayerComponentView elementView = null;
```

Comment:
i think the controller should not be in charge of managing the view's pool.. the responsability should be limited to do something like `voiceChatWindowView.ShowPlayer(new VoiceChatPlayerComponentModel(....))`. The pool and the complexity of the creation should be managed inside the view

Reply:
Hmm... I'm not so sure about that. For me, a view should be a "silly" layer that simply provides different methods to interact with it. The view should be prepared to work using "object pooling" or without that, it simply does what its consumers (the controller in this case) request: add an item, remove an item, set an specific visual state, etc..

I think the way on how the controller want to load the items in the view (using object pooling in this case) is a business logic and shouldn't be a responsibility of the view.

Anyway I think it is an interesting point and I would like to know more opinions about this. Lets ask before and decide what to do, ok?

@AjimenezDCL what do you think about this? Other opinions like yours would be useful for us. Thanks!

Reply:
IMO using or not pools is an implementation detail for the view. You could have multiple implementations, one instantiating/destroying, one using pools, one just caching as many elements as the size of the window fit and reuse them on scrolling....

All of them could share the same controller. This would allow easier scalability for other platforms. In webGL you would have just X entries that are reused and the scrolling would be slower with a loading spinner and in desktop you could have a pool with hundreds of them. This would become messier if you have to implement different controllers.

Reply:
I hadn't thought about it from that point of view of the different platforms... Ok, then I'm going to move the responsibility to the view. Thanks you both guys! ❤️

Reply:
Done in https://github.com/decentraland/unity-renderer/pull/2346/commits/c29679b930fcd4b924c32b38956780627158559c.